XRTK-Core/Packages/com.xrtk.core/Definitions/Controllers/Simulation/SimulatedControllerDataProviderProfile.cs

```diff
@@ -0,0 +1,227 @@
+        [Tooltip("Enable controller simulation")]
+        private bool controllerSimulationEnabled = false;
+
+        /// <summary>
+        /// Is hand simulated hand tracking enabled?
+        /// </summary>
+        public bool ControllerSimulationEnabled => controllerSimulationEnabled;
+
+        [SerializeField]
+        [Implements(typeof(IMixedRealitySimulatedController), TypeGrouping.ByNamespaceFlat)]
+        [Tooltip("The concrete controller type to simulate.")]
+        private SystemType simulatedControllerType = null;
+
+        /// <summary>
+        /// The concrete controller type to simulate.
+        /// </summary>
+        public SystemType SimulatedControllerType
+        {
+            get => simulatedControllerType;
+            private set => simulatedControllerType = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Simulated update frequency for tracking data in milliseconds. 0ms is every frame.")]
+        private double simulatedUpdateFrequency = 0;
+
+        /// <summary>
+        /// The simulated update frequency in milliseconds mimicks the hardware's ability to
+        /// update controller tracking data. A value of 0ms will provide data
+        /// updates every frame.
+        /// </summary>
+        public double SimulatedUpdateFrequency => simulatedUpdateFrequency;
+
+        [SerializeField]
+        [Tooltip("Time after which uncontrolled controllers are hidden")]
+        private float controllerHideTimeout = 0.2f;
+
+        /// <summary>
+        /// ime after which uncontrolled controllers are hidden
+        /// </summary>
+        public float ControllerHideTimeout => controllerHideTimeout;
+
+        #endregion
+
+        #region Placement Settings
+
+        [Header("Placement Settings")]
+
+        [SerializeField]
+        [Tooltip("Default distance of the controller from the camera")]
+        private float defaultDistance = 0.5f;
+
+        /// <summary>
+        /// Default distance of the controller from the camera.
+        /// </summary>
+        public float DefaultDistance => defaultDistance;
+
+        [SerializeField]
+        [Tooltip("Depth change when scrolling the mouse wheel")]
+        private float depthMultiplier = 0.1f;
+
+        /// <summary>
+        /// Depth change when scrolling the mouse wheel.
+        /// </summary>
+        public float HandDepthMultiplier => depthMultiplier;
+
+        [SerializeField]
+        [Tooltip("Apply random offset to the controller position")]
+        private float jitterAmount = 0.0f;
+
+        /// <summary>
+        /// Apply random offset to the controller position.
+        /// </summary>
+        public float JitterAmount => jitterAmount;
+
+        #endregion
+
+        #region Controls Settings
+
+        [Header("Controls Settings")]
+
+        [SerializeField]
+        [Tooltip("Key to toggle persistent mode for the left controller")]
+        private KeyCode toggleLeftPersistentKey = KeyCode.T;
```

Comment:
I'm not a fan of this. I'd rather see this setup as input actions.

Reply:
I know. Same as with editor controls. But that's IMO not in the scope of this PR. I am not seeing how to do that with the current way input actions are done.

Reply:
Agreed, let's revisit in the next iteration.  Also, any change that requires InputActions should be done after the new inputactions PR

Reply:
This PR is for the final feature. If anything I don't want a feature to go in until it's ready.

Reply:
I have no issues with implementing this using Input Actions. But in that case I'll need some rough sketch what you are imagining how to do this. Because I don't see how to do it in a proper way at this point.

Reply:
Do you want me to specify all the required input in the default input actions profile? Where do I map defined input actions to actual input via keyboard etc.? Is it string based?

So e.g. in the simulation profile I'd have to create a string setting "Scroll" to listen for the "Scroll" input action in the data provider? I am not getting how all the input actions thing is supposed to work.